core/src/main/scala/cats/collections/Heap.scala

```diff
@@ -308,11 +308,7 @@ object Heap {
     override def unbalanced: Boolean = size < (1L << height) - 1L
   }
 
-  private[collections] case object Leaf extends Heap[Nothing] {
-    def apply[A](): Heap[A] = this.asInstanceOf[Heap[A]]
-
-    def unapply[A](heap: Heap[A]): Boolean = heap.isEmpty
-
+  final private[collections] case class Leaf[A] private () extends Heap[A] {
```

Comment:
Pretty much the same fix as it was in `Diet.scala` for the "match may not be exhaustive" warnings.